### PR TITLE
NWN2: Wrap up the Trap script functionality

### PR DESCRIPTION
**Pull request by rjshae**

### CODE REVIEW

File: src/engines/nwn2/script/functions_object.cpp [hunk 1]
```diff
@@ -409,7 +409,100 @@ void Functions::getTrapBaseType(Aurora::NWScript::FunctionContext &ctx) {
...
-	ctx.getReturn() = trigger->getTrapBaseType();
+	ctx.getReturn() = (trigger) ? trigger->getTrapBaseType() : 0;
+}
+
+void Functions::getTrapCreator(Aurora::NWScript::FunctionContext &ctx) {
+	NWN2::Object *object = NWN2::ObjectContainer::toObject(getParamObject(ctx, 0));
+	if (!object)
+		return;
+
+	Trigger *trigger = NWN2::ObjectContainer::toTrigger(object);
```
[the remark this answers is not in the document]
Comment (rjshae) [reply src/engines/nwn2/script/functions_object.cpp]: Unless I'm mistaken, that's pretty much true of all the trap script calls, so I should just do a mass revision and assume it's safe to call toTrigger() and check for null?

[the remark this answers is not in the document]
Comment (DrMcCoy) [reply src/engines/nwn2/script/functions_object.cpp]: Yes, please.

**Comment by rjshae** [comment]
Hopefully those changes addressed your concerns. Thanks.

**Comment by DrMcCoy** [comment]
Merged as 6a6214984...9eef8aa3e, thanks! :)

**Pull request closed without merging**